diff --git a/Run.py b/Run.py
index d24aee6..9d9c555 100644
--- a/Run.py
+++ b/Run.py
@@ -45,7 +45,7 @@ if __name__ == '__main__':
 
     # Loading NN model
     logging.info('Loading NN Model...')
-    net = MobileNet_V2(pretrained = True, image_channels=1, num_classes=num_classes)
+    net = ResNet34(pretrained = True, image_channels=1, num_classes=num_classes)
     logging.info(f'Model loaded as "{net.name}"')
     logging.info(f'Network:\n'
                  f'\t{net.in_channels} input channels\n'
diff --git a/utils/__pycache__/tensorboard_logger.cpython-39.pyc b/utils/__pycache__/tensorboard_logger.cpython-39.pyc
index c41c90c..e44db9c 100644
Binary files a/utils/__pycache__/tensorboard_logger.cpython-39.pyc and b/utils/__pycache__/tensorboard_logger.cpython-39.pyc differ
