diff --git a/ResNet/__pycache__/resnet_model.cpython-39.pyc b/ResNet/__pycache__/resnet_model.cpython-39.pyc
index 0ef4a8e..6889906 100644
Binary files a/ResNet/__pycache__/resnet_model.cpython-39.pyc and b/ResNet/__pycache__/resnet_model.cpython-39.pyc differ
diff --git a/ResNet/resnet_model.py b/ResNet/resnet_model.py
index d56a63e..f6281be 100644
--- a/ResNet/resnet_model.py
+++ b/ResNet/resnet_model.py
@@ -124,7 +124,7 @@ class ResNet(torch.nn.Module):
         )
 
         self.avgpool = torch.nn.AdaptiveAvgPool2d((1, 1))
-        self.fc = torch.nn.Linear(512 * 4 + 1, num_classes)
+        self.fc = torch.nn.Linear(512 * 2 + 1, num_classes)
 
     def forward(self, x):
         y = x[1]
