diff --git a/utils/__pycache__/wandb_logger.cpython-39.pyc b/utils/__pycache__/wandb_logger.cpython-39.pyc
index 4cd466b..06eca6c 100644
Binary files a/utils/__pycache__/wandb_logger.cpython-39.pyc and b/utils/__pycache__/wandb_logger.cpython-39.pyc differ
diff --git a/utils/wandb_logger.py b/utils/wandb_logger.py
index eddf262..04ad643 100644
--- a/utils/wandb_logger.py
+++ b/utils/wandb_logger.py
@@ -16,6 +16,7 @@ def wandb_setup(config) -> wandb:
     run_name = config['name']
     device = config['device']
     # Create a run
+    wandb.tensorboard.patch(root_logdir="./tensorboard/run")
     experiment = wandb.init(
         project = "Bone-Age-RSNA", 
         entity = "rsna-bone-age", 
