diff --git a/ResNet/__pycache__/resnet_model.cpython-39.pyc b/ResNet/__pycache__/resnet_model.cpython-39.pyc
index 0ef4a8e..2cb567a 100644
Binary files a/ResNet/__pycache__/resnet_model.cpython-39.pyc and b/ResNet/__pycache__/resnet_model.cpython-39.pyc differ
diff --git a/Train.py b/Train.py
index 4839d00..496fc34 100644
--- a/Train.py
+++ b/Train.py
@@ -128,7 +128,7 @@ def train_net(net, device, train_loader, val_loader,
 
 
                 # Evaluation round
-                division_step = (n_train // (100 * batch_size))
+                division_step = (n_train // (10 * batch_size))
                 if division_step > 0 and global_step % division_step == 0:
                     histograms = {}
                     for tag, value in net.named_parameters():
@@ -142,12 +142,12 @@ def train_net(net, device, train_loader, val_loader,
                     logging.info(f'Validation Dice score: {val_score}, Correct: {correct}')
                     experiment.log({
                         'learning rate': optimizer.param_groups[0]['lr'],
-                        'validation Dice': val_score,
+                        'validation Loss': val_score,
                         'validation Correct': correct,
                         'images': wandb.Image(images[0].cpu()),
                         'Age': {
-                            'True': boneage[0].float().cpu(),
-                            'Pred': age_pred[0].argmax(dim=1, keepdim=True)[0].float().cpu(),
+                            'True': boneage.float().cpu(),
+                            'Pred': age_pred.argmax(dim=1, keepdim=True)[0].float().cpu(),
                         },
                         'step': global_step,
                         'epoch': epoch,
@@ -222,7 +222,7 @@ if __name__ == '__main__':
                  f'\t{net.n_channels} input channels\n'
                  f'\t{net.num_classes} output channels (classes)\n')
     
-    learning_rate = 0.0001
+    learning_rate = 0.01
     epochs = 10
     batch_size = 1
     val_percent = 0.1
diff --git a/checkpoints/INTERRUPTED.pth b/checkpoints/INTERRUPTED.pth
index aef01a0..d5c4b32 100644
Binary files a/checkpoints/INTERRUPTED.pth and b/checkpoints/INTERRUPTED.pth differ
diff --git a/checkpoints/checkpoint_epoch1.pth b/checkpoints/checkpoint_epoch1.pth
index 875157e..2cd426a 100644
Binary files a/checkpoints/checkpoint_epoch1.pth and b/checkpoints/checkpoint_epoch1.pth differ
