diff --git a/ResNet/__pycache__/resnet_model.cpython-39.pyc b/ResNet/__pycache__/resnet_model.cpython-39.pyc
index 0ef4a8e..e42fa1a 100644
Binary files a/ResNet/__pycache__/resnet_model.cpython-39.pyc and b/ResNet/__pycache__/resnet_model.cpython-39.pyc differ
diff --git a/Train.py b/Train.py
index 4839d00..79b6b29 100644
--- a/Train.py
+++ b/Train.py
@@ -187,7 +187,7 @@ def data_organizer(train_dataset, test_dataset, batch_size: int, val_percent: fl
     return train_loader, val_loader, test_loader
 
 def ResNet50(img_channel = 3, num_classes = 1000):
-    return ResNet(Block, [3, 4, 6, 3], img_channel, num_classes, name="ResNet50")
+    return ResNet(Block, [3, 4, 4, 3], img_channel, num_classes, name="ResNet50")
 
 def ResNet101(img_channel = 3, num_classes = 1000):
     return ResNet(Block, [3, 4, 23, 3], img_channel, num_classes, name="ResNet101")
